pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -581,9 +583,22 @@ func (pl *dynamicResources) PreScore(ctx context.Context, cs *framework.CycleSta
 }
 
 func haveAllNodes(nodeNames []string, nodes []*v1.Node) bool {
-	for _, node := range nodes {
-		if !haveNode(nodeNames, node.Name) {
-			return false
+	if len(nodeNames) < len(nodes) {
+		return false
+	}
+	// The time complexity is O(n)
+	if len(nodeNames) > haveAllNodesThreshold {
+		m := sets.New(nodeNames...)
```

Comment:
This line looks suspicious to me. It looks like `podScheduling.Spec.PotentialNodes` could be a set instead of list, which would allow to drop this. From other point of view it's already part of the API. Dunno. @pohly WDYT?

Reply:
The API convention is to use a slice with "list type: set". I am not sure whether it is okay to deviate from that convention in cases like this. It's worth asking.

Reply:
ok, makes sense